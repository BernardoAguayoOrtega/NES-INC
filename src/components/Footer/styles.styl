@import "../../stylus/variables.styl";

.footer
  color: #FFFFFF
  background: linear-gradient(0deg, #4D7080, #325A6C)
  display: flex
  width: 100%
  height 300px
  flex-direction: column
  justify-content: space-evenly
  align-items: center
  bottom: 0
  margin-top: 10px
  &__social--icons
    width: 35px
    background-color: transparent
    border-radius: 30%
    padding: 4px
    border: 2px solid black
    margin: 15px 10px 0px
  &__subtitle
    display: flex
    justify-content: space-around
    font-weight: 200
    font-family: Font1
    width: 30%
    margin-bottom: 20px
    a
      text-decoration: none
      color: white
  &__subtitle h2
    font-family: Font1
    font-weight: 500
    font-size: 2rem
  &__comments
    text-align: center
    font-weight: 400
    font-family: Font2
  &__logo
    margin: 0
    width: 250px
    height: 120px
    