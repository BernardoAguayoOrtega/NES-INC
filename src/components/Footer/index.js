import React from 'react';
const Facebook = 'https://www.flaticon.es/svg/static/icons/svg/733/733605.svg';
const Twitter = 'https://www.flaticon.es/svg/static/icons/svg/733/733635.svg';
const Instagram = 'https://www.flaticon.es/svg/static/icons/svg/87/87390.svg';
import './styles.styl';


const Footer = () => (
    <footer className='footer'>
        <div className='footer__social'>
            <img className='footer__social--icons' src={Facebook} alt="Facebook"/>
            <img className='footer__social--icons' src={Twitter} alt="Twitter"/>
            <img className='footer__social--icons' src={Instagram} alt="Instagram"/>
        </div>
        <h1 className='footer__title'>
            PLATZI NEWS
        </h1>
        <div className='footer__subtitle'>
            <h2>
                About
            </h2>
            <h2>
                Team
            </h2>
        </div>
            <p className='footer__comments'>
                2020 by Masters INC. Proudly created with React JS
            </p>
    </footer>
);

export default Footer;