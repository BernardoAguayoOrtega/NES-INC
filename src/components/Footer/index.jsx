// import react
import React from 'react';
// import icons
const Facebook = 'https://www.flaticon.es/svg/static/icons/svg/733/733605.svg';
const Twitter = 'https://www.flaticon.es/svg/static/icons/svg/733/733635.svg';
const Instagram = 'https://www.flaticon.es/svg/static/icons/svg/87/87390.svg';
// import logo
import Logo from '../../assets/Logo'
// import styles
import './styles.styl';

const Footer = () => (
  <footer className="footer">
    <div className="footer__social">
      <img className="footer__social--icons" src={Facebook} alt="Facebook"/>
      <img className="footer__social--icons" src={Twitter} alt="Twitter"/>
      <img className="footer__social--icons" src={Instagram} alt="Instagram"/>
    </div>
    <Logo className="footer__logo"/>
    <div className="footer__subtitle">
      <h2>
        About
      </h2>
      <h2>
        Team
      </h2>
    </div>
    <p className="footer__comments">
      2020 by Masters INC. Proudly created with React JS
    </p>
  </footer>
);

export default Footer;
