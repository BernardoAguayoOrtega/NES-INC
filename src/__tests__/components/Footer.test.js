// import react
import React from 'react';
// import shallow
import { shallow, mount } from 'enzyme';
// import router
import { HashRouter as Router } from 'react-router-dom';
// import toJson
import toJson from 'enzyme-to-json';
// import app component
import Footer from '../../components/Footer/index.jsx';

describe('<Footer />', () => {
  const wrapper = shallow(<Footer />);

  it('is render the Footer?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('matches the snap shot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('<Footer /> mount', () => {
  const wrapper = mount(
    <Router>
      <Footer />
    </Router>,
  );

  it('is render the Footer?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('are there a Footer social  tag?', () => {
    expect(wrapper.find('.footer__social').children().length).toBe(3);
  });

  it('should Footer have the correct text?', () => {
    expect(wrapper.find('.footer__comments').text()).toBe(
      '2020 by Masters INC. Proudly created with React JS',
    );
  });

  it('should Footer have two about link component?', () => {
    expect(wrapper.find('Link').first().prop('to')).toBe('/about');
  });

  it('should Footer link component has the correct route?', () => {
    expect(wrapper.find('Link').last().prop('to')).toBe('/Privacy_Policy');
  });
});
