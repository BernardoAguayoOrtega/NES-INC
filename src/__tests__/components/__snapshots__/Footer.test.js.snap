// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches the snap shot 1`] = `
<footer
  className="footer"
>
  <div
    className="footer__social"
  >
    <img
      alt="Facebook"
      className="footer__social--icons"
      src="https://www.flaticon.es/svg/static/icons/svg/733/733605.svg"
    />
    <img
      alt="Twitter"
      className="footer__social--icons"
      src="https://www.flaticon.es/svg/static/icons/svg/733/733635.svg"
    />
    <img
      alt="Instagram"
      className="footer__social--icons"
      src="https://www.flaticon.es/svg/static/icons/svg/87/87390.svg"
    />
  </div>
  <Logo
    className="footer__logo"
  />
  <div
    className="footer__subtitle"
  >
    <Link
      className="about-link"
      to="/about"
    >
      <h2>
        About
      </h2>
    </Link>
    <Link
      className="team-link"
      to="/team"
    >
      <h2>
        Team
      </h2>
    </Link>
    <Link
      className="policy-link"
      to="/Privacy_Policy"
    >
      <h2>
        Privacy Policy
      </h2>
    </Link>
  </div>
  <p
    className="footer__comments"
  >
    2020 by Masters INC. Proudly created with React JS
  </p>
</footer>
`;
